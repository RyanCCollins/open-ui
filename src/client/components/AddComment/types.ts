import * as React from 'react';
import AddComment from './';

export interface AddCommentProps extends React.Props<typeof AddComment> {
  input: string;
  onSubmit: Function;
  onChange: Function;
  onKeyUp: Function;
}
