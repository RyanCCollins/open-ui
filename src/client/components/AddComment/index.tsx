import * as React from 'react';
import { AddCommentProps } from './types';
import {
  Wrapper,
  CommentWrapper,
  PicWrapper,
  Pic,
  TextArea,
  Footer,
  Actions,
  Action,
  ActionButton,
} from './styles';

export default function AddComment({
  onSubmit,
  input,
  onChange,
  onKeyUp,
}: AddCommentProps): JSX.Element {
  return (
    <Wrapper>
      <PicWrapper>
        <Pic />
      </PicWrapper>
      <CommentWrapper>
        <form>
          <TextArea
            onKeyUp={onKeyUp}
            value={input}
            onChange={onChange}
            name="comment-input"
            id="comment-input"
            cols="30"
            rows="3"
            placeholder="Add comment..."
          />
        </form>
        <Footer>
          <Actions>
            <Action onClick={onSubmit}>
              <ActionButton>Submit</ActionButton>
            </Action>
          </Actions>
        </Footer>
      </CommentWrapper>
    </Wrapper>
  );
};
