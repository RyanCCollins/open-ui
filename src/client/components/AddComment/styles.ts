const styled = require('styled-components').default;

export const Wrapper = styled.div`
  min-height: 5.3125rem;
  margin-bottom: 1.25rem;
  width: 100%;
  display: flex;
  flex-direction: row;
`;

export const CommentWrapper = styled.div`
  padding: 1rem;
  background-color: #fff;
  vertical-align: top;
  border-radius: 0.1875rem;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.08);
  flex: 1;
`;

export const Footer = styled.footer`
  color: #acb4c2;
  font-size: 0.875rem;
  display: flex;
  align-items: baseline;
  justify-content: flex-end;
  flex-direction: row;
`;

export const Actions = styled.ul`
  list-style: none;
`;

export const Action = styled.li`
  padding-right: 0.75rem;
`;

export const ActionButton = styled.a`
  cursor: pointer;
`;

export const PicWrapper = styled.div`
  padding-top: 0.625rem;
  width: 3.5rem;
`;

export const Pic = styled.div`
  height: 2.25rem;
  width: 2.25rem;
  border-radius: 50%;
  background-size: contain;
  background-image: url('https://github.com/RyanCCollins/cdn/blob/master/misc/missing.png?raw=true');
`;

export const TextArea = styled.textarea`
  width: 100%;
  resize: none;
  outline: none;
  border: none;
  display: block;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  font-family: "PT Sans", "Helvetica Neue", "Helvetica", "Roboto", "Arial", sans-serif;
  font-size: 1rem;
  color: #555f77;
`;
