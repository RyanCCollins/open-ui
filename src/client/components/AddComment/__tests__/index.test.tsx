import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import AddComment from '../';
import { addCommentProps } from './__mocks__/addCommentMocks.mock';

describe('<AddComment /> rendering', () => {
  it('should render with supplied props', () => {
    const wrapper = shallow(
      <AddComment {...addCommentProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('<AddComment /> onSubmit event testing', () => {
  let addCommentComponent;

  beforeEach(() => {
    addCommentComponent = mount(
      <AddComment {...addCommentProps} />,
    );
  });

  it('AddComment requires onSubmit prop', () => {
    expect(addCommentComponent.props().onSubmit).toBeDefined();
  });

  it('AddComment renders an anchor tag for OnClick event', () => {
    const renderedAnchorTag = addCommentComponent.find('a').first();
    expect(renderedAnchorTag).toBeDefined();
  });

  it('Clicking anchor tag calls onSubmit()', () => {
    const renderedAnchorTag = addCommentComponent.find('a').first();
    renderedAnchorTag.simulate('click');
    expect(addCommentProps.onSubmit).toBeCalled();
  });
});

describe('<AddComment /> onChange event testing', () => {
  let addCommentComponent;

  beforeEach(() => {
    addCommentComponent = mount(
     <AddComment {...addCommentProps} />,
    );
  });

  it('AddComment requires onChange prop', () => {
    expect(addCommentComponent.props().onChange).toBeDefined();
  });

  it('AddComment renders textarea for OnChange event', () => {
    const renderedTextArea = addCommentComponent.find('textarea').first();
    expect(renderedTextArea).toBeDefined();
  });

  it('Changing textarea triggers OnChange()', () => {
    const renderedTextArea = addCommentComponent.find('textarea').first();
    renderedTextArea.simulate('change');
    expect(addCommentProps.onChange).toBeCalled();
  });
});

describe('<AddComment /> onKeyUp event testing', () => {
  let addCommentComponent;

  beforeEach(() => {
    addCommentComponent = mount(
      <AddComment {...addCommentProps} />,
    );
  });

  it('AddComment requires onKeyUp prop', () => {
    expect(addCommentComponent.props().onKeyUp).toBeDefined();
  });

  it('AddComment renders textarea for OnKeyUp event', () => {
    const renderedTextArea = addCommentComponent.find('textarea').first();
    expect(renderedTextArea).toBeDefined();
  });

  it('KeyUp in textarea triggers OnKeyUp()', () => {
    const renderedTextArea = addCommentComponent.find('textarea').first();
    renderedTextArea.simulate('keyup');
    expect(addCommentProps.onKeyUp).toBeCalled();
  });
});
