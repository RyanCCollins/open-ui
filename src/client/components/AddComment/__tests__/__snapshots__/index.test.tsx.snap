// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddComment /> rendering should render with supplied props 1`] = `
<styled.div>
  <styled.div>
    <styled.div />
  </styled.div>
  <styled.div>
    <form>
      <styled.textarea
        cols="30"
        id="comment-input"
        name="comment-input"
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Add comment..."
        rows="3"
        value="Mock input for test"
      />
    </form>
    <styled.footer>
      <styled.ul>
        <styled.li
          onClick={[Function]}
        >
          <styled.a>
            Submit
          </styled.a>
        </styled.li>
      </styled.ul>
    </styled.footer>
  </styled.div>
</styled.div>
`;
